File: JS/Tema4_Ejercicio1.js
/*
 * TEMA 4: ESTRUCTURAS REPETITIVAS - EJERCICIO 1
 * ==============================================
 * Imprimir números del 1 al 10 usando while.
 * 
 * EXPLICACIÓN:
 * - Bucle while: ejecuta mientras condición sea verdadera
 * - Control manual con variable contador
 * - Incremento manual del contador
 */

console.log("=== BUCLE WHILE (1 al 10) ===");
let contador = 1;

while (contador <= 10) {
    console.log(contador);
    contador++;
}

// Versión inversa
console.log("\n=== BUCLE WHILE (10 al 1) ===");
contador = 10;

while (contador >= 1) {
    console.log(contador);
    contador--;
}

/*
 * SALIDA ESPERADA:
 * === BUCLE WHILE (1 al 10) ===
 * 1
 * 2
 * 3
 * 4
 * 5
 * 6
 * 7
 * 8
 * 9
 * 10
 * 
 * === BUCLE WHILE (10 al 1) ===
 * 10
 * 9
 * 8
 * 7
 * 6
 * 5
 * 4
 * 3
 * 2
 * 1
 */

File: Java/Tema4_Ejercicio1.java
/*
 * TEMA 4: ESTRUCTURAS REPETITIVAS - EJERCICIO 1
 * ===============================================
 * Escriba un programa que imprima los números del 1 al 10 usando while.
 */

public class Tema4_Ejercicio1 {
    public static void main(String[] args) {
        System.out.println("=== NÚMEROS DEL 1 AL 10 (WHILE) ===");
        
        int i = 1;
        while (i <= 10) {
            System.out.println("Número: " + i);
            i++;
        }
    }
}

/*
 * SALIDA ESPERADA:
 * === NÚMEROS DEL 1 AL 10 (WHILE) ===
 * Número: 1
 * Número: 2
 * ... (hasta el 10)
 */
